spmd/compiler/fusion.py

```diff
@@ -78,20 +103,44 @@ def update_info(self, gm: fx.GraphModule) -> None:
 
 
 def _insert_fusion_buffer_node(
-    gm: fx.GraphModule, insert_before_node: fx.Node, buffer_size: Iterable[int]
+    gm: fx.GraphModule,
+    buffer_size: int,
+    gi: Optional[GraphInfo] = None,
 ) -> fx.Node:
-    """insert a torch.empty node in front of insert_before_node"""
+    """insert a torch.empty node for the global buffer.
```

Comment:
qq should this come right before the fused op (instead of right after the placeholder) -- to avoid memory overhead? (just a qq, not blocker for this pR) >

Reply:
great point - I was thinking of forwards pass and it would not matter re: memory pressure, but you are right.  
For backwards activations are being freed as it progresses, so we should hold off on creating the global buffer until first use to reduce peak mem pressure.
Not hard to change, but will require some reordering as we have to do it after determining fusion plan...thus prefer to get this PR in as is, and then update so I have made this tracking issue for it:
https://github.com/pytorch/tau/issues/654